src/plugins/irc-events/error.js

```diff
@@ -2,21 +2,57 @@ var Msg = require("../../models/msg");
 
 module.exports = function(irc, network) {
 	var client = this;
-	irc.on("errors", function(data) {
+
+	// TODO: remove later
+	irc.on("irc_error", function(data) {
+		console.log("Got an irc_error");
+		irc.emit("error", data);
+	});
+
+	irc.on("error", function(data) {
+		console.log("error", data);
+		var text = data.error;
+		if (data.reason) {
+			text = data.reason + " (" + text + ")";
+		}
 		var lobby = network.channels[0];
 		var msg = new Msg({
 			type: Msg.Type.ERROR,
-			text: data.message,
+			text: text,
 		});
 		client.emit("msg", {
 			chan: lobby.id,
 			msg: msg
 		});
-		if (!network.connected) {
-			if (data.cmd === "ERR_NICKNAMEINUSE") {
-				var random = irc.me + Math.floor(10 + (Math.random() * 89));
-				irc.nick(random);
-			}
-		}
+	});
+
+	irc.on("nick in use", function(data) {
+		var lobby = network.channels[0];
+		var msg = new Msg({
+			type: Msg.Type.ERROR,
+			text: "Nickname " + data.nick + " is already in use: " + data.reason,
```

Comment:
Testing on Freenode, I get:
```
Nickname astorije is already in use: Nickname is already in use.
```

Is the `reason` something defined by the network admins, by irc-framework, or a standard (*de facto* or not) in IRC?
I feel like we'll have `Nickname is already in use` most of the time, which doesn't look too nice.
What do you think about using the retrieved reason unless there isn't one? Something like:

```js
text: data.reason || "Nickname " + data.nick + " is already in use."
```

A bug I noticed though: if the nickname is already taken, `/nick` fails but the user still gets assigned a new random nickname, instead of not changing nickname.